sachem/Controllers/RechercheInscriptionController.cs

```diff
@@ -0,0 +1,164 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Cryptography;
+using System.Web;
+using System.Web.Mvc;
+using sachem.Models;
+using System.Net;
+using static sachem.Classes_Sachem.ValidationAcces;
+using System.Data.Entity;
+
+namespace sachem.Controllers
+{
+    public class RechercheInscriptionController : Controller
+    {
+        private readonly SACHEMEntities db = new SACHEMEntities();
+
+        const int BROUILLON = 1;
+        const int ACCEPTE = 3;
+        const int REFUSE = 5;
+
+        [ValidationAccesSuper]
+        public ActionResult Index()
+        {
+            var touteInscription = from inscription in db.Inscription
+                                   where inscription.id_Statut != BROUILLON
+                                   select inscription;
+
+            return View(Rechercher());
+        }
+
+        [ValidationAccesSuper]
+        public ActionResult Details(int? id)
+        {
+            if (id == null)
+            {
+                return new HttpStatusCodeResult(HttpStatusCode.BadRequest);
+            }
+
+            Inscription inscriptionPersonne = db.Inscription.Find(id);
+            if(inscriptionPersonne == null)
+            {
+                return HttpNotFound();
+            }
+
+            List<Inscription> inscription = db.Inscription.Where(x => x.id_Pers == inscriptionPersonne.id_Pers).ToList();
```

Comment:
C'est au départ une question de visibilité.
Un bon coté d'utilisé le typage dynamique c'est que ton language reste strongly typed, mais t'as pas besoins de te casser la tête à savoir ce qui va sortir d'une requete Linq top complexe. Linq peut retourner des IEnumerable, IQueryable, List. Tes retour peuvent être generic ou typé.
Le nom de ta variable que tu Query est assez significatif pour savoir ce que ça retourne la plupart du temps.
Le polymorphisme devient pas mal plus facile à gérer, t'as pas besoins de te cassé la tête a savoir si c'est quoi le parents, s'tu une classe abstraite?

Mais, y'a des mauvais coté: Le temps de compilation est plus grand. L'auto-completion de ton IDE ne fonctionne pas dans certain IDE/language. Ton code va être plus maintenable à long terme. Il y a moins de cohésion au sein de l'équipe dans le desing/style.
Mais tous ça peut être palier en utilisant les bon design patern et en faisant du TDD.

dynamic typing = fast to code. for small to medium projects.
static typing = better to maintaint. for huge projects.

et finalement: j'ai trouvé toute tes réponse avec un simple recherche google.
https://www.google.ca/search?q=advantage+of+synamic+typing&oq=advantage+of+synamic+typing&aqs=chrome..69i57j0.4256j0j7&sourceid=chrome&ie=UTF-8
Stackoverflow et les sites comme ça c'est une mine d'or quand t'as des questions comme ça ;)